ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PushQueryPublisher.java

```diff
@@ -93,7 +95,10 @@ public synchronized void subscribe(final Flow.Subscriber<Collection<StreamedRow>
 
       queryMetadata.setLimitHandler(this::setDone);
       queryMetadata.setUncaughtExceptionHandler(
-          (thread, e) -> setError(e)
+          e -> {
+            setError(e);
+            return StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.SHUTDOWN_CLIENT;
```

Comment:
For Push queries should we retry or just shutdown?

Reply:
Do you have any motivations to let push queries just shutdown upon errors here?

Reply:
Today for transient queries we don't retry, and instead just return the error as the final message in the response. I think we should keep the status quo for this patch (which it looks like you're doing)